Database/Posts.hs

```diff
@@ -37,26 +38,27 @@ postQuery conn whereClause args = fmap toPost <$> quickQuery' conn query args
                         , "order by posts.id desc"
                         ]
 
-getPost :: Database -> ID -> IO (Maybe ResolvedPost)
-getPost Database{dbConn} idPost = listToMaybe <$>
-  postQuery dbConn "where posts.id = ?" [toSql idPost]
+getPost :: ID -> DatabaseM (Maybe ResolvedPost)
+getPost idPost = listToMaybe <$>
+  postQuery "where posts.id = ?" [toSql idPost]
 
-postChildren :: Database -> ID -> IO [ResolvedPost]
-postChildren Database{dbConn} idPost =
-  postQuery dbConn "where posts.id_parent = ?" [toSql idPost]
+postChildren :: ID -> DatabaseM [ResolvedPost]
+postChildren idPost =
+  postQuery "where posts.id_parent = ?" [toSql idPost]
 
-getPostsSince :: Database -> Maybe ID -> IO [ResolvedPost]
-getPostsSince Database{dbConn} Nothing = postQuery dbConn "" []
-getPostsSince Database{dbConn} (Just idPost) =
-  postQuery dbConn "where posts.id > ?" [toSql idPost]
+getPostsSince :: Maybe ID -> DatabaseM [ResolvedPost]
+getPostsSince Nothing = postQuery "" []
+getPostsSince (Just idPost) =
+  postQuery "where posts.id > ?" [toSql idPost]
 
-insertPost :: Database -> User -> Text -> Maybe ID -> UTCTime -> IO (Maybe ResolvedPost)
-insertPost db@(Database{dbConn, dbPostChan}) User{userID = idUser} content idParent at =
-  insertRow dbConn query args >>= maybe (return Nothing) report
+insertPost :: User -> Text -> Maybe ID -> UTCTime -> DatabaseM (Maybe ResolvedPost)
+insertPost User{userID = idUser} content idParent at =
+  insertRow query args >>= maybe (return Nothing) report
   where
     report idPost = do
-      post <- fromJust <$> getPost db idPost
-      writeChan dbPostChan post
+      db <- ask
+      post <- fromJust <$> getPost idPost
+      liftIO $ writeChan (dbPostChan db) post
```

Comment:
This is gross, because I'm writing to the sockets from within the database layer. A future change will completely separate them.